backend/pkg/rest/applications_resources_controller.go

```diff
@@ -71,3 +73,162 @@ func (s *Server) GetApplicationResources(params operations.GetApplicationResourc
 			Total: &total,
 		})
 }
+
+func (s *Server) DeleteApplicationResource(id string) middleware.Responder {
+	// Get application resource info.
+	resource, err := s.dbHandler.ResourceTable().GetDBResource(id, false)
+	if err != nil {
+		log.Errorf("application resource not found by id %q: %v", id, err)
+		return operations.NewDeleteApplicationResourcesIDNotFound()
+	}
+
+	// Get package IDs that has relations with the resource.
+	relatedPackageIDs, err := s.getRelatedPackageIDs(resource.ID)
+	if err != nil {
+		log.Errorf("Failed to get packages for resource ID=%s: %v", resource.ID, err)
+		operations.NewDeleteApplicationResourcesIDNotFound()
+	}
+
+	// Delete only applications<->resources<->packages relationships.
+	if err = s.dbHandler.JoinTables().DeleteRelationships(database.DeleteRelationshipsParams{
+		ResourceIDsToRemove: []string{resource.ID},
+	}); err != nil {
+		log.Error(fmt.Sprintf("Failed to delete application relationships: %v", err))
+		return operations.NewDeleteApplicationResourcesIDDefault(http.StatusInternalServerError).
+			WithPayload(oopsResponse)
+	}
+
+	// Get list of package IDs that has no any relations to other resources.
+	packageIDsToDelete, err := s.getPackagesToDelete(relatedPackageIDs)
+	if err != nil {
+		log.Errorf("Failed to get remaining packages: %v", err)
+		operations.NewDeleteApplicationResourcesIDNotFound()
+	}
+
+	// Get list of vulnerability IDs for the packages that needs to be deleted.
+	relatedVulnerabilityIDs, err := s.getRelatedVulnerabilityIDs(packageIDsToDelete)
+	if err != nil {
+		log.Errorf("Failed to get vulnerabilities for packages: %v", err)
+		return operations.NewDeleteApplicationResourcesIDDefault(http.StatusInternalServerError).
+			WithPayload(oopsResponse)
+	}
+
+	// Delete only package<->vulnerabilities relationships.
+	if err = s.dbHandler.JoinTables().DeleteRelationships(database.DeleteRelationshipsParams{
+		PackageIDsToRemove: packageIDsToDelete,
+	}); err != nil {
+		log.Errorf("Failed to delete application relationships: %v", err)
+		return operations.NewDeleteApplicationResourcesIDDefault(http.StatusInternalServerError).
+			WithPayload(oopsResponse)
+	}
+
+	// Get list of vulnerability IDs that have no relationship for existing packages.
+	vulnerabilityIDsToDelete, err := s.getVulnerabilitiesToDelete(relatedVulnerabilityIDs)
+	if err != nil {
+		log.Errorf("Failed to get remaining vulnerabilities: %v", err)
+		return operations.NewDeleteApplicationResourcesIDDefault(http.StatusInternalServerError).
+			WithPayload(oopsResponse)
+	}
+
+	if err = s.dbHandler.ResourceTable().Delete(resource); err != nil {
+		log.Errorf("Failed to delete application from DB: %v", err)
+		return operations.NewDeleteApplicationResourcesIDDefault(http.StatusInternalServerError).
+			WithPayload(oopsResponse)
+	}
+
+	s.deletePackages(packageIDsToDelete)
+	s.deleteVulnerabilities(vulnerabilityIDsToDelete)
```

Comment:
maybe we want to delete the related application if not connected to another resource?

Reply:
We can use an existing application and attach new scans for it. WDYT @akpsgit

Reply:
I don't think that we should. If the user created it manually and then scanned an image and deleted it, i wouldn't expect that the user need to add the application again. Maybe it's more relevant for automatically created applications, but let's leave it out. WDYT?